File: Assets/Scripts/ObjectControl.js
#pragma strict

public var speed : float = 10;
public var dirRight : boolean = false;


function FixedUpdate() {
	var dir : Vector3 = Vector3(1, 0, 0);
	if (!dirRight){
		dir.x = -1;
	}
	transform.Translate(dir * speed * Time.deltaTime);
}

File: Assets/Scripts/_ObjectControl.cs
using UnityEngine;

public class _ObjectControl : MonoBehaviour {
	[SerializeField] private float _velocity;
	
	private SpriteRenderer spriteRenderer;
	
	public float velocity {
		get { return _velocity; }
		set { _velocity = value; }
	}
	
	public bool flip {
		get { return spriteRenderer.flipX; }
		set { spriteRenderer.flipX = value; }
	}
	
	private void Awake() {
		spriteRenderer = this.GetComponent<SpriteRenderer>();
	}

	private void Update() {
		Vector2 translation = !spriteRenderer.flipX ? Vector2.left : Vector2.right;
		transform.Translate(translation * _velocity * Time.deltaTime);
	}
	
	private void OnBecameInvisible() {
		Destroy(gameObject);
	}
}
